compiler/frontend/src/org/jetbrains/kotlin/resolve/lazy/FileScopeFactory.kt

```diff
@@ -214,20 +217,24 @@ class FileScopeFactory(
 
             scope = currentPackageScope(packageView, aliasImportNames, dummyContainerDescriptor, FilteringKind.INVISIBLE_CLASSES, scope)
 
-            scope = LazyImportScope(
-                scope, defaultAllUnderImportResolver, defaultLowPriorityImportResolver, LazyImportScope.FilteringKind.VISIBLE_CLASSES,
-                "Default all under imports in $debugName (visible classes)"
-            )
+            if (createDefaultImportingScopes) {
+                scope = LazyImportScope(
+                    scope, defaultAllUnderImportResolver, defaultLowPriorityImportResolver, LazyImportScope.FilteringKind.VISIBLE_CLASSES,
+                    "Default all under imports in $debugName (visible classes)"
+                )
+            }
 
             scope = LazyImportScope(
                 scope, allUnderImportResolver, null, LazyImportScope.FilteringKind.VISIBLE_CLASSES,
                 "All under imports in $debugName (visible classes)"
             )
 
-            scope = LazyImportScope(
-                scope, defaultExplicitImportResolver, null, LazyImportScope.FilteringKind.ALL,
-                "Default explicit imports in $debugName"
-            )
+            if (createDefaultImportingScopes) {
+                scope = LazyImportScope(
```

Comment:
does this scope include explicit imports from the same line? If yes, why it is not needed in repl when we have imports in the line?

Reply:
Yes, but in this case `createDefaultImportingScopes` is `true`, see implementation of `org.jetbrains.kotlin.scripting.compiler.plugin.repl.ReplCodeAnalyzerBase.ResettableAnalyzerState#computeFileScopes`

Reply:
Sorry, I may be missing something again, but I don't see any relevant call to the `createScopesForFile` with `createDefaultImportingScopes = true`.

Reply:
Sorry for misleading, I've just haven't understood you correctly. Default imports **do not** include explicit imports. But for the first snippet in session it is called with `true`: in `org.jetbrains.kotlin.resolve.repl.ReplState#submitLine`, in override of `org.jetbrains.kotlin.resolve.lazy.FileScopesCustomizer#createFileScopes`.